### PR TITLE
feat(web): create tag on the fly

### PR DESCRIPTION
**Pull request by MehringTing**
https://github.com/immich-app/immich/discussions/12179#discussion-7115039

### CODE REVIEW

**Comment by alextran1502** [comment]
Hello, thank you for the PR. Can you help fix the e2e test?

File: web/src/lib/components/shared-components/combobox.svelte [hunk 1]
```diff
@@ -221,9 +225,15 @@
     searchQuery = selectedOption ? selectedOption.label : '';
   });
 
-  let filteredOptions = $derived(
-    options.filter((option) => option.label.toLowerCase().includes(searchQuery.toLowerCase())),
-  );
+  let filteredOptions = $derived.by(() => {
+    const _options = options.filter((option) => option.label.toLowerCase().includes(searchQuery.toLowerCase()));
+
+    if (allowCreate && searchQuery !== '' && _options.filter((option) => option.label === searchQuery).length === 0) {
```
Comment (danieldietzler) [inline web/src/lib/components/shared-components/combobox.svelte:231]: This filter doesn't do anything does it? You only want `_options.length === 0`

File: web/src/lib/components/shared-components/combobox.svelte [hunk 2]
```diff
@@ -36,6 +36,8 @@
     options?: ComboBoxOption[];
     selectedOption?: ComboBoxOption | undefined;
     placeholder?: string;
+    allowCreate?: boolean;
+    defaultFirstOption?: boolean;
```
Comment (alextran1502) [inline web/src/lib/components/shared-components/combobox.svelte:40]: Can you help put some comment of what does this props do?

File: web/src/lib/components/forms/tag-asset-form.svelte [hunk 3]
```diff
@@ -50,20 +55,22 @@
 <FullScreenModal title={$t('tag_assets')} icon={mdiTag} onClose={onCancel}>
   <div class="text-sm">
     <p>
-      <FormatMessage key="tag_not_found_question">
+      <!-- <FormatMessage key="tag_not_found_question">
```
Comment (alextran1502) [inline web/src/lib/components/forms/tag-asset-form.svelte:58]: Can you remove this unused code?

**Review by alextran1502 — COMMENTED** [review]
I just tested the PR and it works well.

Can you help address the comments? And what do you think about making a wrapper component for the combobox and adding specific tag creation handling in that wrapper component instead of adding those logics to the combobox?

**Comment by MehringTing** [comment]
> I just tested the PR and it works well.
> 
> Can you help address the comments? And what do you think about making a wrapper component for the combobox and adding specific tag creation handling in that wrapper component instead of adding those logics to the combobox?

I refer to the design of the element plus for the select component.

image

**Review by jrasm91 — APPROVED** [review]
LGTM

**Review by danieldietzler — APPROVED** [review]

**Pull request merged**